### ISSUE TITLE
Squiz.Arrays.ArrayDeclaration.ValueNoNewline incorrectly detects nested arrays

### ISSUE THREAD

**Issue by mhujer** [squizlabs/PHP_CodeSniffer#383]
`Each value in a multi-line array must be on a new line` error is detected for `array` on the line 7 in the following snippet (which I think is formatted properly):

```
class Foo
{
    public function getDecorators()
    {
        return array(
            'ViewHelper',
            array(  // <--- error here
                'Foo',
            ),
            'Errors',
        );
    }
}
```

**Comment by mhujer** [comment]
I might have found another issue - with array comments. When I want to add comments to array, this way is reported as incorrect (*Each value in a multi-line array must be on a new line*):
```
$ids = [
    '1', //foo
    '13', //bar
];
```

I found only this not very nice workaround:
```
$ids = [
    '1' /* foo */,
    '13' /* bar */,
];
```

Do you have

**Comment by gsherwood** [comment]
The Squiz standard doesn't allow comments to appear after statements, so that array sniff has never accounted for them. But I'm going to change it to ignore comments after the statement so your first code block doesn't throw any errors as this will make the sniff more generic.

**Comment by gsherwood** [comment]
I've committed the fix for this. Thanks for reporting.

**Comment by mhujer** [comment]
Thanks for fixing it
